x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/AuditTrail.java

```diff
@@ -81,4 +82,9 @@ void runAsDenied(String requestId, Authentication authentication, RestRequest re
     void explicitIndexAccessEvent(String requestId, AuditLevel eventType, Authentication authentication, String action, String indices,
                                   String requestName, TransportAddress remoteAddress, AuthorizationInfo authorizationInfo);
 
+    // this is the only audit method that is called *after* the action executed, when the response is available
+    // it is however *only called for coordinating actions*, which are the actions that a client invokes as opposed to
+    // the actions that a node invokes in order to service a client request
+    void coordinatingActionResponse(String requestId, Authentication authentication, String action, TransportRequest transportRequest,
+                                    TransportResponse transportResponse);
```

Comment:
This is not technically true, some actions' (`TransportMasterNodeAction`) responses are intercepted on the coordinating node, as well as on the master node.

The naming and descriptions are difficult to write to convey the subset of actions that are covered.
We could also bleed some internals in here and mention the `SecurityActionFilter`.

Open to suggestions.

Reply:
I think we could keep the previous name, i.e. `actionResponse`. Please let me explain: When I suggested pulling out the code from `SecurityServerTransportInterceptor`, it was in the context of getting auditing for API key ID in for v7.11.0. The reasons are:
* The response in `SecurityServerTransportInterceptor` is a fairly low level indexing response and does not really provide extra valuable information than the one in the coordinating node. Therefore auditing on just the coordinating node is sufficient.
* Creating API key is underlyingly a `TransportReplicationAction`. Unlike `TransportMasterNodeAction`, it does _not_ go through the `SecurityActionFilter` (on the remote node). This means: on the remote node, we will not get another auditing entry from `SecurityActionFilter`, i.e. there will only be a single auditing entry about API key creation and it is on the coordinating node.

So my motivation was rather narrowly focused on the particular use case. The agreement, as I understand it, is to defer the auditing code in `SecurityServerTransportInterceptor` until the next release. Because it is not necessary for API key auditing in this release and if it is not in use, it's better to be left out. But we also agreed that it could still be useful when we expand the usage scenario and this is subject to further discussion. 

IIUC, your intention of changing the method name is for the possibility to have two different audit event types. I should have been clear that it was not my proposal. I consider whether auditing inside `SecurityServerTransportInterceptor` an implementation detail, which we chose to opt out for initial release, but could choose to add it later as an enhancement. For verbosity and duplication control, I'd prefer to solve it in a different way other than splitting it into different audit event types. I hope this makes sense.

Reply:
Thanks for the detailed explanation Yang, but I would like to focus the conversation on the scope of this PR only. We should consider the broader picture, as you describe, but ultimately this PR is what gets merged as a unit, and what we merge in the future is subject to debate still.

If we merge this PR, which adds the audit hook for the responses of actions intercepted by the `SecurityActionFilter` only, we add to the interface of the `LoggingAuditTrail` .
When we later merge the code to allow intercepting the responses of the other actions, we have the options to:
* reuse the same `LoggingAuditTrail` hook method
* rename the existing hook method, or
* add a new audit response hook method.
(or any mix of them, really, if we feel wild)

Because this is a private interface, I don't believe we have to name it in anticipation of its use. We should name it for the purpose that it serves right now, as we can always tinker with it later. For this reason, I would not favor the `actionResponse` name.

Reply:
Just trying to brainstorm here: How about calling this one `actionResponse` and the other one `transportResponse`?
As you said, this is a private interface, so we don't have to struggle too much with its name now. So please feel free to use whatever works for you.

Reply:
Yeah, this could work. I'll keep this in mind when we'll iterate on the interface.